src/main/java/org/example/CalendarManagement/calendarfacade/MeetingFacade.java

```diff
@@ -0,0 +1,121 @@
+        if(employeesNotAvailable.size()>0){
+            response = new Response("these employees are not available", employeesNotAvailable);
+            return response;
+        }
+
+        Integer meetingRoomId = 0;
+        if(addMeetingDataRequest.getRoomName().equals(""))
+        {
+            int officeId = employeeRepository.findOfficeIdById(addMeetingDataRequest.getOwnerId());
+            meetingRoomId =  meetingRoomService.findFreeMeetingRoom(officeId,addMeetingDataRequest.getDateOfMeeting(),addMeetingDataRequest.getStartTime(),addMeetingDataRequest.getEndTime());
+
+            if(meetingRoomId == 0)
+            {
+                response = new Response("No Meeting room is available", "");
+                return response;
+            }
+            response = new Response(null, meetingRoomId);
+        }
+        else
+        {
+            boolean meetingRoomAvailableResponse = meetingRoomService.meetingRoomAvailable(addMeetingDataRequest.getRoomName(),
+                    addMeetingDataRequest.getDateOfMeeting(),addMeetingDataRequest.getStartTime(),addMeetingDataRequest.getEndTime());
+
+            if(!meetingRoomAvailableResponse)
+            {
+                response = new Response("Given Meeting room is not available","");
+                return response;
+            }
+            Optional<MeetingRoom> meetingRoom = meetingRoomRepository.findByName(addMeetingDataRequest.getRoomName());
+            meetingRoomId = meetingRoom.get().getRoomId();
+            response = new Response(null, meetingRoom.get().getRoomName());
+        }
+
+        MeetingDetails meetingDetails = new MeetingDetails();
+        meetingDetails.setDescription(addMeetingDataRequest.getDescription());
+        meetingDetails.setAgenda(addMeetingDataRequest.getAgenda());
+        meetingDetails.setOwnerId(addMeetingDataRequest.getOwnerId());
+        meetingDetails.setDateOfMeeting(new Date(addMeetingDataRequest.getDateOfMeeting().getDayOfMonth(),addMeetingDataRequest.getDateOfMeeting().getMonthValue(),addMeetingDataRequest.getDateOfMeeting().getYear()));
+        meetingDetails.setStartTime(new Time(addMeetingDataRequest.getStartTime().getHour(),addMeetingDataRequest.getStartTime().getMinute(),addMeetingDataRequest.getStartTime().getSecond()));
+        meetingDetails.setEndTime(new Time(addMeetingDataRequest.getEndTime().getHour(),addMeetingDataRequest.getEndTime().getMinute(),addMeetingDataRequest.getEndTime().getSecond()));
+        meetingDetails.setIsAvailable(true);
+        meetingDetails.setRoomId(meetingRoomId);
+
+        String meetingId = null;
+
+        try {
+            meetingId = meetingServiceClient.addMeetingDetails(meetingDetails);
+            response = new Response(null, meetingId);
+        }catch (RuntimeException ex){
+            throw new RuntimeException(ex.getMessage());
+        }
+
+        List<EmployeeStatusDataRequest> employeeStatusDataRequestArrayList = new ArrayList<EmployeeStatusDataRequest>();
```

Comment:
see if this can be clubed to addmeetingDetails

Reply:
clubbed